arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/nullable.kt

```diff
@@ -0,0 +1,10 @@
+package arrow.core.continuations
+
+@Suppress("ClassName")
+public object nullable {
+  public inline fun <A> eager(crossinline f: suspend OptionEagerEffectScope.() -> A): A? =
+    option.eager(f).orNull()
```

Comment:
there is one issue I found is problematic in the signature of `A?.bind():A` in OptionEffectScope or the eager version. It is allows this:
```kotlin
// context(EffectScope<DomainError>)
val a: Int? = null
a.bind().bind().bind()
```
without the compiler or Intellij able to grey out that the other two `bind()` are not necessary. I've added a contract for that case here, but still won't highlight it. Am I missing something?

Reply:
These kinds of accesses are not detected by the compiler but by the IDEA plugin. The plugin will tell you `"".toString().toString()"` is redundant but it does not know `.bind().bind()` is redundant because it has no knowledge of bind(). It only knows these https://github.com/JetBrains/intellij-kotlin/blob/e2c105911e1fca5b7fe6b9dc10cf9f75463a3c13/idea/src/org/jetbrains/kotlin/idea/intentions/RemoveRedundantCallsOfConversionMethodsIntention.kt#L36-L51 

Making it universal it's not possible, some operators with other semantics may perform side effects and it's legit to chain their calls:
```kotlin
acquire().acquire()
release(2)
```

Reply:
Thanks Raul, makes sense.

Reply:
Right, that's why we had a separate `nullable` DSL that worked *only* for `nullable` types.
Otherwise `A?.bind()` took precedence over `Option<A>.bind()` 🤔

I still think the `nullable` DSL is useful enough to keep around though, but then we should keep it separate again.
We made this mistake in the past and had to fix it through deprecations and removal. This should be traceable by checking the git history of the Option and Nullable computation DSL.